packages/@lwc/ssr-runtime/src/lightning-element.ts

```diff
@@ -72,19 +65,19 @@ export class LightningElement implements PropsAvailableAtConstruction {
     }
 
     [SYMBOL__SET_INTERNALS](props: Properties, attrs: Attributes) {
+        this.#props = props;
         this.#attrs = attrs;
         assign(this, props);
+    }
+
+    get className() {
+        return this.#props.class ?? '';
+    }
 
-        defineProperty(this, 'className', {
-            get() {
-                return props.class ?? '';
-            },
-            set(newVal) {
-                props.class = newVal;
-                attrs.class = newVal;
-                mutationTracker.add(this, 'class');
-            },
-        });
+    set className(newVal: any) {
+        this.#props.class = newVal;
+        this.#attrs.class = newVal;
+        mutationTracker.add(this, 'class');
```

Comment:
Avoid calling `defineProperty` on-the-fly – just use the same pattern for `#props` that we already do for `#attrs`.